src/query/sql/src/executor/physical_plan.rs

```diff
@@ -72,6 +73,35 @@ impl TableScan {
     }
 }
 
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct IndexTableScan {
+    /// A unique id of operator in a `PhysicalPlan` tree.
+    /// Only used for display.
+    pub plan_id: u32,
+
+    pub name_mapping: BTreeMap<String, IndexType>,
+    pub source: Box<DataSourcePlan>,
+
+    /// Only used for display
+    pub table_index: IndexType,
+    pub stat_info: Option<PlanStatsInfo>,
+
+    pub internal_column: Option<BTreeMap<FieldIndex, InternalColumn>>,
+}
+
```

Comment:
The structure is totally the same as `TableScan`, if we can reuse `TableScan` but add a new field to indicate it is for aggregating index?